.github/workflows/test.yml

```diff
@@ -4,29 +4,103 @@ on: [push, pull_request]
 
 env:
   CARGO_TERM_COLOR: always
+  # CI builds are closer to a from-scratch build anyways
+  # and incremental compilation hurts reproducibility
+  CARGO_INCREMENTAL: 0
 
 jobs:
-  test:
+  unit-tests:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
         os: [macos-latest, ubuntu-latest]
     steps:
     - uses: actions/checkout@v2
-    - name: Unit tests
+
+    - uses: Swatinem/rust-cache@v1.3.0
+
+    - name: Build kak-lsp
+      run: cargo test --no-run --locked --release
+
+    - name: Run unit tests
       run: cargo test --release
-    - name: Install system test dependencies
+
+  integration-tests:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [macos-latest, ubuntu-latest]
+    steps:
+    - uses: actions/checkout@v2
+
+    - uses: Swatinem/rust-cache@v1.3.0
+
+    - name: Build kak-lsp
+      run: cargo build --locked --release
+
+    - name: Fetch Kakoune
+      uses: actions/checkout@v2
+      with:
+        repository: 'mawww/kakoune'
+        ref: 'v2021.08.28'
```

Comment:
Shouldn't we keep using HEAD to find issues before a Kakoune release? This could cause intermittent failures that are not kak-lsp's fault but the Kakoune master branch has been fairly stable.